user-fund-storage/src/fund_transf_module.rs

```diff
@@ -52,45 +52,59 @@ pub trait FundTransformationsModule {
         Ok(())
     }
 
-    fn swap_waiting_to_active<I: Fn() -> bool>(&self, amount: &BigUint, interrupt: I) -> (Vec<usize>, BigUint) {
+    fn swap_waiting_to_active<I: Fn() -> bool>(&self, amount: &BigUint, interrupt: I, dry_run: bool) -> (Vec<usize>, BigUint) {
         let mut stake_to_activate = amount.clone();
-        let mut affected_states = self.fund_module().split_convert_max_by_type(
+        let mut affected_users: Vec<usize> = Vec::new();
+        self.fund_module().split_convert_max_by_type(
             Some(&mut stake_to_activate),
             FundType::Waiting,
-            |_, _| Some(FundDescription::Active),
-            interrupt
+            SwapDirection::Forwards,
+            |user_id, _| {
+                affected_users.push(user_id);
```

Comment:
this way the code for computing the affected users is complicated. it is easier to read the way it was before - when split_convert_max_by_type to return the affected users.

Reply:
You are computing affected users also in many places where you don't need it. This costs extra gas/performance

Reply:
This pattern is pretty typical lambda usage, this is how the for_each method works. Since we already have access to the user_id in the lambda, it is redundant to also return the list of user_ids separately.

Reply:
for_each pattern in Rust: https://docs.rs/foreach/0.3.0/foreach/